### ISSUE TITLE
Improve contributor guide and developer instructions

### ISSUE THREAD

**Issue by elmps2018** [opendatakit/tool-suite-X#223]
We appreciate the many folks who (want to) contribute to the ODK-X software! We currently have a read me for each repository on github (e.g. https://github.com/odk-x/app-designer) and one about contributing to code in [docs](https://docs.odk-x.org/advanced-topics-developer/). There's also this one specifically about [contributing to the docs](https://docs.odk-x.org/docs-tech-guide/) 

This is an ongoing and open issue. If you find something outdated/confusing about setting up a development environment and fix it refer to this issue!

**Comment by iamhks** [comment]
Hi @elmps2018 !! 
I would like to work on this issue.

**Comment by elmps2018** [comment]
Hi @iamhks! This is an open issue that anyone could work on! I have assigned it to you but if you can comment with what, specifically, you end up working on other folks can then contribute as well!

**Comment by Kanak1081** [comment]
Hi Team,
I would also like to contribute for the same and will update about the changes done 😊

**Comment by elmps2018** [comment]
Hi @Kanak1081! Great! Assigning to you as well. Please coordinate a bit with @iamhks to collaborate or divide-and-conquer!

**Comment by wbrunette** [comment]
related issue: https://github.com/odk-x/tool-suite-X/issues/229

**Comment by melibleq** [comment]
Hi eveyrone. I'm Melissa, from São Paulo - Brazil. I've been selected for the Outreachy contribution phase and ODK-X project #1 sounds interesting.
I tried joining the Slack channel, but it shows me a message saying that the link isn't active anymore.
Is there still a way for me to join it? I would like to learn more and hopefully countribute with good first issues.
I'm sorry if this message bothers you, but I still wanted to reach out for help. Please let me know if there's a better way to carry on with our conversation.
Thanks a lot in advance!
@wbrunette @linl33 @elmps2018

**Comment by iamhks** [comment]
Hi @melibleq ! 

> I tried joining the Slack channel, but it shows me a message saying that the link isn't active anymore.
> I'm sorry if this message bothers you, but I still wanted to reach out for help. Please let me know if there's a better way to carry on with our conversation.

I don't think it's necessary to apologize for reaching out.
Secondly, please check this for joining the slack channel: https://forum.odk-x.org/t/slack-link-show-not-active-anymore/1304/2
p.s The forum is active too.

**Comment by melibleq** [comment]
Hey there @iamhks, thanks a lot! I've just joined it. Signing up for the forum too.

**Comment by OyinOlamide** [comment]
Hello @elmps2018 , I am an Outreachy applicant and would like to work on this. Can I please be assigned to this?

**Comment by elmps2018** [comment]
Hi @OyinOlamide! This is an open issue that anyone could work on! I have assigned it to you but if you can comment with what, specifically, you end up working on other folks can then contribute as well!

**Comment by AliceUmeji** [comment]
Hello @wbrunette, @elmps2018 may I work on this issue? 

While working on fixing a typo I discovered that the number 5 of 'Working on the Docs' section (first line) simply read 'Finally, you can open an editor of your choice and work on the documentation.' without detailing out the steps needed to accomplish this.

What I mean is that the document should assume that the person reading it is a complete newbie and list out steps to take in order to make the edit in the text editor (VSCode in my case). Anyone else who is familiar with other text editors can also contribute in that.

If this is ok with you, I'd like you to point in the direction I should take to handle it (I think I should work on it differently and the link will be embedded into that section or what do you think?)

Thank you.

**Comment by wbrunette** [comment]
Yes you may work on this issue. I think it's probably best to provide a link that others will update over time. Maybe a wikipedia link https://en.wikipedia.org/wiki/Source-code_editor

**Comment by AliceUmeji** [comment]
Ok @wbrunette, will look into how to do that. Thank you

**Comment by wbrunette** [comment]
@AliceUmeji  - I should also clarify that what I mean was add a link that points to another website that is constantly being updated over time. (I didn't mean that others would update ODK-X docs; instead, we get to take advantage of the world updating wikipedia).

**Comment by AliceUmeji** [comment]
Oh @wbrunette you mean I should edit wikipedia and embed its link in ODK-X docs right? Hope this still counts as an Outreachy contribution please

**Comment by wbrunette** [comment]
Adding a sentence about Code editors and then a link to popular ones count as a contribution.

**Comment by AliceUmeji** [comment]
@wbrunette Ok, thank you. Will let you know when I'm done.

**Comment by AliceUmeji** [comment]
@wbrunette, during my research I found that you can edit directly in github. I believe this is easier and more straight forward. Please is there a reason why you don't use this method? or should I include this method instead? It can be added as a tip.

**Comment by wbrunette** [comment]
@AliceUmeji We do mention this method (maybe not in the contribution guide) so feel free to add it as a tip. The problem is that you do not get the errors you need to fix without running it locally. So it's only helpful for minor changes. With regards to Outreachy one of the reasons we do documentation as starting contribution is because the applicant is demonstrating skills using various tools. Checking out code from GitHub, setting up your environment, etc.

**Comment by harloma** [comment]
I would like to work on this issue, can I get assigned to this

**Comment by elmps2018** [comment]
Hi @harloma! I assigned you - but wanted to note this is something that other folks can work on as well. Let us know if you have any questions while working away.

**Comment by harloma** [comment]
Thanks
